pkg/manifest/subscription.go

```diff
@@ -138,6 +155,17 @@ func subscriptionService(subscriptionOptions subscription.ImageOptions, serviceO
 			commands = append(commands, "/usr/bin/insights-client --register")
 			// insights-client creates the .gnupg directory during boot process, and is labeled incorrectly
 			commands = append(commands, "restorecon -R /root/.gnupg")
+			// register to template if template is specified
+			if subscriptionOptions.TemplateUUID != "" {
+				patchURL := subscriptionOptions.PatchURL
+				if patchURL[len(patchURL)-1] == '/' {
+					patchURL = patchURL[:len(patchURL)-1]
+				}
+				addTemplateURL := fmt.Sprintf("%s/templates/%s/subscribed-systems", patchURL, subscriptionOptions.TemplateUUID)
+				curlToAssociateSystem := fmt.Sprintf("curl -v --retry --proxy %s %s --cert /etc/pki/consumer/cert.pem  --key /etc/pki/consumer/key.pem -X PATCH", subscriptionOptions.Proxy, addTemplateURL)
```

Comment:
I just noticed that `-proxy` is always added to the command. However, I expect that specifying the proxy is not mandatory. Would it make more sense to add it to the command conditionally, in case `subscriptionOptions.Proxy` is not empty?

Reply:
agreed, that is likely the better practice